core/src/main/java/org/elasticsearch/action/support/ThreadedActionListener.java

```diff
@@ -81,7 +81,7 @@ public ThreadedActionListener(Logger logger, ThreadPool threadPool, String execu
         this.logger = logger;
         this.threadPool = threadPool;
         this.executor = executor;
-        this.listener = listener;
+        this.listener = ContextPreservingActionListener.wrap(listener, threadPool.getThreadContext(), false);
```

Comment:
I know we talked about it but reflecting on this, I think it's too implicit to wrap things here. Better be explicit and put this as a parameter type and let the caller be aware and decide.